src/Dependencies.jl

```diff
@@ -57,13 +58,33 @@ is_runtime_dependency
 Define a binary dependency that is necessary to build the package and load the
 generated JLL package.  The argument can be either a string with the name of the
 JLL package or a `Pkg.PackageSpec`.
+
+The optional keyword argument `build_version` can be used to specify the version
+of the dependency to be installed when building it.
+
+The optional keyword argument `compat` can be used to specify a string for
+use in the `Project.toml` of the generated Julia package.
 """
 struct Dependency <: AbstractDependency
     pkg::PkgSpec
     build_version::Union{VersionNumber,Nothing}
-    Dependency(pkg::PkgSpec, build_version = nothing) = new(pkg, build_version)
+    compat::String  # semver string for use in Project.toml of the JLL
+    function Dependency(pkg::PkgSpec, build_version = nothing; compat::String = "")
```

Comment:
I almost wonder if we should just not allow `PkgSpec` objects anymore.  We won't ever look up a package outside of `General`, so I'm not sure what the use of a `PkgSpec` is anymore....

Reply:
Maybe for backwards-compatibility, we can define a converter method that takes in a `PkgSpec`, extracts just the name, then passes that to the "true" `Dependency()` constructor, which stores only the name.

Reply:
@giordano do you think we should change `Dependency` to store `name::String` instead of `pkg::PkgSpec`?

Reply:
Interesting. That would indeed simplify some things (also make the code easier to understand, IMHO).

Naively thought, this seems fine to me. However, one then looses the ability to also specify a UUID. I've never been a fan of that in `build_tarballs.jl` scripts, *but* perhaps there are reasons why being able to do so might be useful? Like, those supply chain attacks some weeks ago against big companies, where attackers registered public packages with names identical to internal only packages, and then internal tooling got confused and picked up those malicious packages instead... That said, if I know the name of an internal package to attack, I might also know its UUID, so I am not sure this would be a valid defense.

I think none of this matter for Yggdrasil, but what about other users?

Reply:
In Yggdrasil we won't build packages outside of General, but users do: `PackageSpec` is useful when you want to build using unregistered dependencies (it's even [documented](https://docs.binarybuilder.org/dev/FAQ/#Can-I-publish-a-JLL-package-locally-without-going-through-Yggdrasil?-1) :slightly_smiling_face:) or dependencies deployed locally.  Also, I can think of users using JLLs in their own registries.

Reply:
So leave in PackageSpec? Fine by me (also: less work ;-))